Version updated to Python 3.9 and significant usability updates

KK1L opened the pull request.

KK1L committed:
Create DS1054Z full version by KK1L

# Jan 2021 KK1L
#   Updated to Python 3.9
# Jul 16 2021 KK1L
#   Added Clear button to remove the waveforms
#   Added toggle button to choose channel 1 or 2 from scope
#   Got the averaging function working and made it do a running average across N bins with no (minimal) execution time hit
#   On screen progress feedback for scope read, FFT, and display
#   Warning trap when scope not triggered

KK1L committed:
Create PyDSA.py

# Jan 2021
# update to Python 3.9

# Jul 16 2021 KK1L
#   Added Clear button to remove the waveforms
#   Added toggle button to choose channel 1 or 2 from scope
#   Got the averaging function working and made it do a running average across N bins with no (minimal) execution time hit
#   On screen progress feedback for scope read, FFT, and display
#   Warning trap when scope not triggered

MichaDit commented:
Thanks for this, it works like a treat 👍🏻
I installed latest nivisa, latest pyvisa (py -m pip install pyvisa) and numpy.
Last I downloaded PyDSA.py directly from your repo - that is all to get it working.

KK1L commented:
I am next going to make the amplitude to be correct in dBm...at least that is the next goal. Then maybe extra waveform storage.I am glad it worked for you. It is my first contribution on GitHub and in Python.I am not sure how to make this a directory in the original location.On Jul 17, 2021 02:58, Fuchks ***@***.***> wrote:

Thanks for this, it works like a treat 👍🏻

I installed latest nivisa, latest pyvisa (py -m pip install pyvisa) and numpy.

Last I downloaded PyDSA directly from your repo - that is all to get it working.



—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

KK1L commented:
I have made some further updates.
- Python 3.9 update...syntax, call format changes, functions
- More robust scope communication
- On screen progress feedback
- Warning if scope is not triggered (used to crash)
- Warning if input signal is clipping at scope
- Can read from channel 1 or channel 2
- Peak detect information and marker on waveform
- Added autoscaling
- Scaling change and offset works for all waves on the screen
- Can store up to 7 waveforms on the screen
- Most actions can be done while acquiring
- Added clear function which removes all waves and resets averaging/max-hold
- Scale is accurate in dBm after calibration
    *Previously the absolute level was arbitrary, but relatively accurate
    *"Calibrate" is manual operation to adjust peak for a known dBm value
    *Use a sine wave of known magnitude and adjust the calibration factor for the right reading
    Note: It is on my list to make this automatic.
